core/fetcher.go

```diff
@@ -0,0 +1,83 @@
+package core
+
+import (
+	"context"
+	"fmt"
+
+	rpctypes "github.com/celestiaorg/celestia-core/rpc/core/types"
+	"github.com/celestiaorg/celestia-core/types"
+	"github.com/celestiaorg/celestia-node/service/block"
+)
+
+const newBlockSubscriber = "NewBlock/Events"
+
+var newBlockEventQuery = types.QueryForEvent(types.EventNewBlock).String()
+
+type BlockFetcher struct {
+	Client
+
+	newBlockCh chan *block.Raw
+}
+
+// NewBlockFetcher returns a new `BlockFetcher`.
+func NewBlockFetcher(client Client) *BlockFetcher {
+	return &BlockFetcher{
+		Client: client,
+	}
+}
+
+// GetBlock queries Core for a `Block` at the given height.
+func (f *BlockFetcher) GetBlock(ctx context.Context, height *int64) (*block.Raw, error) {
+	raw, err := f.Block(ctx, height)
+	return raw.Block, err
+}
+
+// SubscribeNewBlockEvent subscribes to new block events from Core, returning
+// a new block event channel on success.
+func (f *BlockFetcher) SubscribeNewBlockEvent(ctx context.Context) (<-chan *block.Raw, error) {
```

Comment:
I believe there would be multiple uses of Subscription per app. That is, multiple components in the Node will listen for new on the network Blocks. Hence, we need to think of this case and better in advance. 

We can either:
* Have one BlockFetcher instance per Node and multiple callers will call `SubscribeNewBlockEvent` for their own subscription.
* Each caller who needs a subscription should create their own instance of BlockFetcher, but then we need a component who will spawn those. This can be BlockService or other.

We can for now live without this I think, but we should create an issue for this case, referencing the comment, if you don't want to solve this in the PR.

Reply:
> I believe there would be multiple uses of Subscription per app. That is, multiple components in the Node will listen for new on the network Blocks. Hence, we need to think of this case and better in advance.

I disagree -- what other components do you imagine would have to listen for new blocks? There should only be one process on the node that is subscribed to new block events.

Reply:
Yeah, we should optimize for the use-case at hand. I think there should be one central place to fetch blocks from core/app. There are not multiple callers? Or am I missing something?

Reply:
There will be clients of the Node listening for new blocks and some component needs to process all those subscriptions as separate subscriptions, but again we can keep it as it is and come to this later.

Reply:
Can you explain why these components would use mutiple clients directly? This is how I understand it: there will be one client for new blocks talking to the core node. Everything else listening for new blocks will get that from a subsystem running inside the node.